godtools/App/Features/AppLanguage/Presentation/DownloadableLanguages/DownloadableLanguagesViewModel.swift

```diff
@@ -76,14 +114,34 @@ extension DownloadableLanguagesViewModel {
     
     func downloadableLanguageTapped(downloadableLanguage: DownloadableLanguageListItemDomainModel) {
         
+        let languageId = downloadableLanguage.languageId
+        
         switch downloadableLanguage.downloadStatus {
             
         case .notDownloaded:
             
-            downloadToolLanguageUseCase.downloadToolLanguage(downloadableLanguage.languageId)
-                .sink { _ in
+            activeDownloads[languageId] = .downloading(progress: 0)
+            
+            downloadToolLanguageUseCase.downloadToolLanguage(languageId: downloadableLanguage.languageId, languageCode: downloadableLanguage.languageCode)
+                .receive(on: DispatchQueue.main)
+                .sink(receiveCompletion: { [weak self] completed in
                     
-                }
+                    switch completed {
+                    case .finished:
+                        
+                        self?.activeDownloads.removeValue(forKey: languageId)
+                        
+                    case .failure(let error):
+                        
+                        // TODO: - what happens during a failure?
+                        
+                        self?.activeDownloads[languageId] = .notDownloaded
+                        self?.flowDelegate?.navigate(step: .showLanguageDownloadErrorAlert(error: error))
```

Comment:
Hey @levieggertcru, there was one time last week that the service was failing, so I added an alert for the moment so it's obvious when that happens.  Do we know what's supposed to happen instead (do we retry?)?  I can catch up with Jill/Daniel if needed and make a follow-up ticket.

Reply:
Hey @rachaelblue hmm that's a good question. I feel like we should do something here.  I'm curious how @frett is handling languages that fail to download.

Reply:
I feel like we should do something here, maybe provide additional UI that shows the download failed with option to retry.  Let's definitely hold off for a separate ticket on that.  I feel like an alert message could get messy if multiple downloads are failing.

Reply:
I know it's a little bit of a delayed response, but on Android the DownloadManager will transparently retry in the background when it can. My download progress indicator is also linked to the number of translations currently downloaded, so it will show as incomplete until that last translation is downloaded successfully

Reply:
on Android the interaction between this view and the download manager is indirect.

This UI just toggles a setting in the database and displays what the database reports as already downloaded.

The DownloadManager monitors the database for changes, and proactively downloads translations for tools that are marked to be downloaded. Once it finishes downloading a translation it marks that in the database.

So those 2 components both interact with the database, but aren't dependent on each other

Reply:
Thanks @frett for the feedback.  I'm going to create a ticket to have our service do some type of retry.